### PR TITLE
Migration to API 1.2 with more functions

### PR DESCRIPTION
**Pull request by softedco**
Ripped from my GameJolt extension for Scratch with some minor adjustments
Tested it and it should work
Didn't change the minified version

### CODE REVIEW

**Comment by MausGames** [comment]
Thank you, I will try to look at the PR on the weekend or next week, I'm currently quite busy, and need to update the docs and unit-tests.

Though at a quick glance, as far as I know the broken data-store functions are related to a different way POST-requests are signed in v1.2. I have a local fix available for it and will add it after the PR, so we can get rid of the old URL.

**Comment by softedco** [comment]
For me I just removed the '>' character from the url and it seems to have worked, but I just couldn't code it to do that,
.replace() didn't work for me both on the SendRequestEx and __CreateAjax method, so I just went with an old url.

File: source/gj-js-api.js [hunk 1]
```diff
@@ -461,3 +463,110 @@ function __CreateAjax(sUrl, sBodyData, pCallback)
...
+};
+GJAPI.ScoreGetTables = function (pCallback) {
+    GJAPI.SendRequest('/scores/tables/?game_id=' + GJAPI.iGameID, GJAPI.SEND_GENERAL, pCallback);
+}
+GJAPI.SessionCheck = function (pCallback) {
+    GJAPI.SendRequest('/sessions/check/?game_id=' + GJAPI.iGameID + '&username=' + GJAPI.sUserName + '&user_token=' + GJAPI.sUserToken, GJAPI.SEND_GENERAL, pCallback);
+};
+
+/* SessionOpen and SessionClose combined */
+GJAPI.SessionSetStatus = function (isOpen) {
```
Comment (MausGames) [inline source/gj-js-api.js]: Why are the functions GJAPI.SessionSetStatus and GJAPI.UserFetchComb implemented here in the library? 

I think if someone needs a combination of multiple base functions, it should be trivial to do it themselves based on their individual requirements. The code in those functions seems to be duplicated and the base functions can be used directly.

**Comment by MausGames** [comment]
Sorry for the late reply, I thought my comment/review got submitted, but it was only pending until now.

File: source/gj-js-api.js (hunk 1 above)
Comment (softedco) [reply source/gj-js-api.js]: It seems that I forgot to remove these, as I said, I used the code from my scratch extension. Yeah sorry for that, they shouldn't be there

**Comment by MausGames** [comment]
Thank you again for the changes, and sorry for the massive delay.

**Pull request merged**